### PR TITLE
DO NOT MERGE: Readme updates

### PR DESCRIPTION
**Pull request by lincolnthree**

### CODE REVIEW

**Review by geoand — APPROVED** [review]

File: README.adoc [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+
+Run the following command to determine the appropriate URL to access our demo. Make sure you access the url with the HTTP scheme. HTTPS is NOT enabled by default:
+
+```bash
+oc get route spring-boot-circuit-breaker-greeting -o jsonpath='http://{.spec.host}{"\n"}'
+```
+
+The result of the above command is the istio-system istio-ingress URL. Open this URL in your a web browser.
+
+== Circuit breaker without Istio
```
Comment (metacosm) [inline README.adoc]: Why did you change the text here?

File: README.adoc [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+
+```bash
+    oc new-project circuit-breaker-mission
+```
+*CAUTION*:
+In order for Istio automatic sidecar injection to work properly the following Istio configuration needs to be in place:
+
+. The `policy` field is set to `disabled` in the `istio-inject` configmap  of the `istio-system` namespace
+. The `istio-sidecar-injector` `MutatingWebhookConfiguration` should not limit the injection to properly labeled namespaces
+. Expose services and Istio ingress
```
Comment (metacosm) [inline README.adoc]: This is not needed. Why did you add this?

File: README.adoc [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+
+This configuration is used to define service names and deployments that control how pods are labeled/versioned on the OpenShift cluster. Labels and versions are key concepts for creating load-balanced or multi-versioned pods in a service.
+
+
+=== With Source to Image build (S2I)
+Run the following commands to apply and execute the OpenShift templates that will configure and deploy the applications:
+```bash
+find . | grep openshiftio | grep application | xargs -n 1 oc apply -f
+
+oc new-app --template=spring-boot-circuit-breaker-greeting -p SOURCE_REPOSITORY_URL=https://github.com/snowdrop/spring-boot-circuit-breaker-booster  -p SOURCE_REPOSITORY_REF=with-sse -p SOURCE_REPOSITORY_DIR=name-service
```
Comment (metacosm) [inline README.adoc]: This is outdated and incorrect. These parameters have been changed.

File: README.adoc [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+oc new-app --template=spring-boot-circuit-breaker-name -p SOURCE_REPOSITORY_URL=https://github.com/snowdrop/spring-boot-circuit-breaker-booster  -p SOURCE_REPOSITORY_REF=with-sse -p SOURCE_REPOSITORY_DIR=name-service
+```
+
+= Use Cases
+== Access the application
+
+Run the following command to determine the appropriate URL to access our demo. Make sure you access the url with the HTTP scheme. HTTPS is NOT enabled by default:
+
+```bash
+oc get route spring-boot-circuit-breaker-greeting -o jsonpath='http://{.spec.host}{"\n"}'
```
Comment (metacosm) [inline README.adoc]: This is incorrect as well.

File: README.adoc [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
++
+```bash
+oc create -f istio/route_rule.yml -n $(oc project -q)
+```
+. At this point, nothing should have changed in the application behavior.
+
+. Now apply the `DestinationPolicy` that activates Istio's Circuit Breaker on the name service:
++
+```bash
+oc create -f istio/destination_policy.yml -n $(oc project -q)
```
Comment (metacosm) [inline README.adoc]: Incorrect.

**Review by metacosm — CHANGES_REQUESTED** [review]
I think this new version is based off an older version of the README. Please note that the booster has been moved to its own repository and is using `master`, in particular.

File: README.adoc (hunk 1 above)
Comment (geoand) [reply README.adoc]: Looks like it was brought in line with the other boosters?

File: README.adoc (hunk 2 above)
Comment (geoand) [reply README.adoc]: Why is it incorrect? 
If I am reading it correctly, it is correct and it's purpose is to make the user aware of the Istio configuration that needs to be in place for this booster to work

File: README.adoc (hunk 3 above)
Comment (geoand) [reply README.adoc]: I guess the template needs to be `spring-boot-istio-circuit-breaker-greeting` (like in the version of README in master)?

File: README.adoc (hunk 2 above)
Comment (metacosm) [reply README.adoc]: I'm talking about exposing the services and Istio ingress. There is no need to do so for this booster.

Comment (geoand) [reply README.adoc]: Cool, thanks for the clarification!

File: README.adoc (hunk 3 above)
Comment (metacosm) [reply README.adoc]: Basically, the content from `master` should be kept (with maybe reformating)… `SOURCE_REPOSITORY_REF` is also incorrect (should point to `master`). These changes had already been made…

**Comment by metacosm** [comment]
I really think it'd be easier to re-apply the formatting changes to the version that's currently on master than try to fix this PR.

File: README.adoc (hunk 2 above)
Comment (lincolnthree) [reply README.adoc]: I added this because all of the boosters should have the same Istio configuration docs. Even if this step is not necessary for *this* booster, if they go to try another booster and think they've set up Istio correctly, it will fail. Granted, none of this really matters because users should be using Istiooc anyway.

File: README.adoc (hunk 1 above)
Comment (lincolnthree) [reply README.adoc]: Correct. This all needs to be consistent/uniform.

**Comment by lincolnthree** [comment]
Please re-review. Thanks.

File: README.adoc [hunk 6]
```diff
@@ -0,0 +1,152 @@
...
+.. https://istio.io/docs/setup/kubernetes/quick-start.html
+.. https://istio.io/docs/setup/kubernetes/ansible-install.html
+. Enable automatic sidecar injection for Istio (See https://istio.io/docs/setup/kubernetes/sidecar-injection.html[this] for details)
++
+In order for Istio automatic sidecar injection to work properly, the following Istio configuration needs to be in place:
++
+.. The `policy` field is set to `disabled` in the `istio-inject` configmap  of the `istio-system` namespace.
+.. The `istio-sidecar-injector` `MutatingWebhookConfiguration` should not limit the injection to properly labeled namespaces
+
+. Expose services and Istio ingress:
```
Comment (metacosm) [inline README.adoc]: This is still unneeded. There is no need to expose the `istio-ingress` for this booster.

File: README.adoc [hunk 7]
```diff
@@ -0,0 +1,152 @@
...
+mvn clean package fabric8:deploy -Popenshift
+```
+Configuration for FMP may be found both in pom.xml and `src/main/fabric8` files/folders.
+
+=== With Source to Image build (S2I)
+Run the following commands to apply and execute the OpenShift templates that will configure and deploy the applications:
+```bash
+find . | grep openshiftio | grep application | xargs -n 1 oc apply -f
+
+oc new-app --template=spring-boot-istio-circuit-breaker-greeting -p SOURCE_REPOSITORY_URL=https://github.com/snowdrop/spring-boot-istio-circuit-breaker-booster  -p SOURCE_REPOSITORY_REF=master -p SOURCE_REPOSITORY_DIR=name-service
```
Comment (metacosm) [inline README.adoc]: `SOURCE_REPOSITORY_DIR` is wrong, it should say `greeting-service` here.

**Review by metacosm — CHANGES_REQUESTED** [review]
Looks good to me apart the 2 things to change. `index.html` also needs to be updated since the link in the instructions section points to `README.md` and should now point to `README.adoc`.
